lib/internal/errors.js

```diff
@@ -140,9 +140,9 @@ class AssertionError extends Error {
     if (message) {
       super(message);
     } else {
-      if (actual && actual.stack && actual instanceof Error)
```

Comment:
This wasn't a performance microoptimization, but rather an integrity check to avoid `Object.create(Error.prototype)` from being recognized as an Error.

Reply:
It was definitely a micro-optimization, see https://github.com/nodejs/node/pull/15025. It's possible it has other side-effects although I'm not convinced `Object.create(Error.prototype)` is something we should worry about in this particular code.

Reply:
I indeed did not think about this case. I also do not mind if it stays as it will not hurt.

Reply:
@TimothyGu would you like to keep it to be on the safe side about this or shall I just remove the part again?

Reply:
@apapirovski Thanks for digging up the original PR. It was very informative. @BridgeAR Feel free to remove.